### PR TITLE
Backend error reporting/handling and propagation of errors to client

### PR DESCRIPTION
**Pull request by sigurdp**
* Added set of service level exceptions
* Added FastAPI exception handlers that send client response and log the errors so that they can be viewed both in console logs and through app insights
* Start using new service exceptions in SummaryAccess
* Utilize new service exceptions in `AuthenticatedUser`
* In `AuthenticatedUser`, removed the currently unused access tokens for `pdm` and `ssdl`
* Re-implemented `AddProcessTimeToServerTimingMiddleware` as pure ASGI middleware to avoid extra exception levels obfuscating the backtraces

Related to issue #86

Todo:
- [x] Remove test code in `realizations_vector_data` route

### CODE REVIEW

File: backend/src/services/sumo_access/_helpers.py [hunk 1]
```diff
@@ -16,8 +17,11 @@ async def _init_helper(access_token: str, case_uuid: str) -> Tuple[SumoClient, C
     sumo_client: SumoClient = create_sumo_client_instance(access_token)
     case_collection = CaseCollection(sumo_client).filter(uuid=case_uuid)
 
-    if await case_collection.length_async() != 1:
-        raise ValueError(f"None or multiple sumo cases found {case_uuid=}")
+    matching_case_count = await case_collection.length_async()
+    if matching_case_count == 0:
+        raise NoDataError(f"Sumo case not found {case_uuid=}", Service.SUMO)
```
Comment (rubenthoms) [inline backend/src/services/sumo_access/_helpers.py:22]: A bit better readable with `NoDataError(f"Sumo case not found for {case_uuid=}", Service.SUMO)` ?

File: backend/src/services/sumo_access/_helpers.py [hunk 2]
```diff
@@ -16,8 +17,11 @@ async def _init_helper(access_token: str, case_uuid: str) -> Tuple[SumoClient, C
...
+    if matching_case_count > 1:
+        raise MultipleDataMatchesError(f"Multiple sumo cases found {case_uuid=}", Service.SUMO)
```
Comment (rubenthoms) [inline backend/src/services/sumo_access/_helpers.py:24]: Same here.

File: backend/src/services/service_exceptions.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        super().__init__(message_for_telemetry)
+
+        self.message = message
+        self.service = service
+
+    def get_error_type_str(self) -> str:
+        return self.__class__.__name__
+
+    def __str__(self) -> str:
+        return f"SSSSS {self.message} [service={self.service.value}]"
```
Comment (rubenthoms) [inline backend/src/services/service_exceptions.py:25]: Remove `SSSSSS`.

**Review by rubenthoms — CHANGES_REQUESTED** [review]
LGTM! Remove debug code and consider adjusting to comments.

File: backend/src/services/service_exceptions.py (hunk 3 above)
Comment (sigurdp) [reply backend/src/services/service_exceptions.py:25]: Absolutely 👍

File: backend/src/services/sumo_access/_helpers.py (hunk 2 above)
Comment (sigurdp) [reply backend/src/services/sumo_access/_helpers.py:24]: 👍

File: backend/src/services/sumo_access/_helpers.py (hunk 1 above)
Comment (sigurdp) [reply backend/src/services/sumo_access/_helpers.py:22]: 👍

File: backend/src/backend/primary/routers/timeseries/router.py [hunk 4]
```diff
@@ -52,6 +56,19 @@ async def get_realizations_vector_data(
...
+    # TO BE REMOVED BEFORE MERGE
+    if vector_name == "FGIR":
+        raise InvalidDataError("Dummy service error because vector name is FGIR", Service.SUMO)
+    if vector_name == "FGIT":
+        raise AuthorizationError("Dummy service error because vector name is FGIT", Service.SUMO)
+    if vector_name == "FGLIR":
+        raise HTTPException(404, "Dummy HTTP layer error, because vector name is FGLIR")
+    # !!!!!!!!!!!!!!!!!!!!!!!!!!
+    # !!!!!!!!!!!!!!!!!!!!!!!!!!
+
```
[the remark this answers is not in the document]
Comment (sigurdp) [reply backend/src/backend/primary/routers/timeseries/router.py]: Test code has been removed 👍

**Review by rubenthoms — APPROVED** [review]
:+1:

**Pull request merged**